test/input-test.jl

```diff
@@ -0,0 +1,96 @@
+using SpikingNN
+using Statistics
+using Test
+
+@testset "input" begin
+
+    @testset "LIF SpikeCount" begin
+        # LIF params
+        τm = 100
+        vreset = 0.0
+        vth = 0.1
+        R = 1.75
+
+        # Input spike train params
+        rate = 0.05
+        T = 1000
+
+        lif = Neuron(QueuedSynapse(Synapse.Delta()), LIF(τm, vreset, R), Threshold.Ideal(vth))
+        input = ConstantRate(rate)
+
+
+        spikescount = []
+        for i in 1:T
+            spikes = excite!(lif, input, T)
+            push!(spikescount, length(spikes))
+        end
+
+        avgspikecount = mean(spikescount)/T
+
+
+        @test Float32(round(avgspikecount,digits = 2)) == input.rate
+    end
+
+    @testset "SRM SpikeCount" begin
+
+        # SRM0 params
+        η₀ = 5.0
+        τᵣ = 1.0
+        vth = 0.5
+
+        # Input spike train params
+        rate = 0.01
+        T = 1000
+
+        srm = Neuron(QueuedSynapse(Synapse.Alpha()), SRM0(η₀, τᵣ), Threshold.Ideal(vth))
+        input = ConstantRate(rate)
+
+
+        spikescount = []
+        for i in 1:T
+            spikes = excite!(srm, input, T)
+            push!(spikescount, length(spikes))
+        end
+
+        avgspikecount = mean(spikescount)/T
+
+        @test Float32(round(avgspikecount,digits = 2)) == input.rate
+        
+    end

```

Comment:
You don't need to use any neurons like LIF or SRM0 for this test. I was thinking of something much simpler like:
```julia
@testset "ConstantRate" begin
    T = 10_000
    rate = rand()
    input = ConstantRate(rate)
    @test count(x -> x > 0, [input(t) for t in 1:T]) / T ≈ rate
end
```

Reply:
Thanks. Wasn't aware of this. This sure is way too cleaner.